_posts/2017-12-28-Secure-your-architecture-part2.md

```diff
@@ -0,0 +1,211 @@
+$> export SECRET=$(curl -X POST --data-urlencode secret http://user:password@localhost:<port>/encrypt)
+$> echo "datasource.password=$SECRET" >> application-dev.yml
+{% endhighlight %}
+
+When the encryption is done, we get a encrypted piece of data in your configuration in the form of:
+{% highlight yml %}
+spring:
+  datasource:
+    username: dbuser
+    password: '{cipher}FKSAJDFGYOS8F7GLHAKERGFHLSAJ'
+{% endhighlight %}
+
+## What to store where?
+When designing your config server, you have different options on where to store and fetch your sensitive data. 
+
+### Using a Git repository
+The default and most common way most of us use it is via private git repositories where we store our sensitive data where the config server can fetch it.
+Be aware, never put configuration inside your code repository, it violates [the twelve-factor app](https://12factor.net/){:target="_blank"} which requires strict separation of config from code. 
+Config varies substantially across deploys, code does not.
+
+#### Health checks
+You can enable your Config Server to health check your repositories by the minute.
+If you do this, always look at which version control would be the best fit, always check when they go in to maintenance. 
+It could be that they host it in another timezone, which could lead to a cascading failure.
+In my opinion, you can just disable the health checks with `spring.cloud.config.server.health.enabled=false` and avoid further failures. 
+If you expect the config server might go down temporarily when your client app starts, please provide a retry mechanism after a failure. 
+To enable a retry, first add `spring-retry` to your classpath with `@EnableRetry` annotation and `spring.cloud.config.failFast=true`
+
+### Using JDBC
+New to this list is the support for JDBC. This enables us to store configuration properties inside a relation database. 
+By switching the active spring profile to jdbc and adding the dependency of spring-jdbc to your classpath, Spring Boot will configure the datasource you included on the classpath.
+To store the data you will need to set up new tables to your database.
+For more information: [using JDBC](http://cloud.spring.io/spring-cloud-static/spring-cloud-config/1.4.0.RELEASE/single/spring-cloud-config.html#_jdbc_backend){:target="_blank"}
+
+### Using Vault
+[HashiCorp's Vault](https://www.vaultproject.io/){:target="_blank"} provides a centralized external management server. 
+Vault can manage static and dynamic secrets such as username/password for remote applications/resources and provide credentials for external services such as MySQL, PostgreSQL, Apache Cassandra, MongoDB, Consul, AWS and more.
+Spring supports using the [Vault as a backend](http://cloud.spring.io/spring-cloud-static/spring-cloud-config/1.4.0.RELEASE/single/spring-cloud-config.html#_vault_backend){:target="_blank"} for Spring Cloud Config.
+If you are using Spring Boot, a quick way to enable vault is to set your spring profile to vault. 
```

Comment:
Vault
Spring profile to Vault.

Reply:
It should be *to set your spring profile to "vault"*, because it's the actual name of the profile.